### PR TITLE
[TT-1563] Add segment tags on oas api

### PR DESCRIPTION
**Pull request by titpetric**

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2263304305)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: 686bb8fa247fa7ff3c3493a4105f15805d895017 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2263707127)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: 5ca73725a66907c20c38a62d3d13ca72377866b1 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2263962874)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: ef22c49c0821c428e12d6e033b394533751efb11 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2264238998)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: d1ef26813c229252eff3e524690dff5e55d27546 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2264272508)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: 007e784e3111c79c2c15525cfd9d91319f91fc23 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2264388025)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: 83f547e02f204b640d96d816c970a02b79eb1b73 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2264610084)

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -595,6 +595,8 @@ type APIDefinition struct {
 	EnableDetailedRecording    bool                   `bson:"enable_detailed_recording" json:"enable_detailed_recording"`
 	GraphQL                    GraphQLConfig          `bson:"graphql" json:"graphql"`
 	AnalyticsPlugin            AnalyticsPluginConfig  `bson:"analytics_plugin" json:"analytics_plugin"`
+	EnableSegmentTags          bool                   `bson:"enable_segment_tags" json:"enable_segment_tags"`
+	SegmentTags                []string               `bson:"segment_tags" json:"segment_tags"`
```
Comment (jeffy-mathew) [inline apidef/api_definitions.go:599]: we already have `tags` here - https://github.com/TykTechnologies/tyk/pull/4030/files#diff-9961ccc89a48d32db5b47ba3006315ef52f6e5007fb4b09f8c5d6d299c669d67R589

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: 51d914636a3d8e6dc5fb1d86e2f7a14c82fad934 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2268453993)

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: 266b300248b02bdb916d6aeb1ef8c016437dbcd6 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2269496673)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4030/merge
Commit: d5e2a74ab27467ec54e91403d6b392d4ae15de28 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2270285598)

**Review by lghiur — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: b5251f040a1ad429f7ffb197d402f5086fa9ab73 [TT-1563] Add segment tags on oas api (#4030)

* Add gatewayTags schema definition

* Add segmentTags api definition, fill/export from/to OAS

* Bump go-bindata to @master version, use go install

* Add segment tags to tyk schema definition, fix bson/json tag for enabling segment tags

* Use existing 'tags' api definition instead of segment_tags

* Rename flags for segment tags for consistency

* Update generated schema file
Triggered by: push (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2270544500)

**Pull request merged**